### ISSUE TITLE
Unreasonable spline curves

### ISSUE THREAD

**Issue by MengnanCui** [uf3/uf3#14]
Hi, I am trying to fit a UF2 model for SiC, based on Formation energies and forces, the dataset generated by annealing MD calculation 200~ snapshot configurations.
Here is the splines curve I got for 5/200, 20/200, 40/200, and 80/200 training datasets, respectively.
![image](https://user-images.githubusercontent.com/28955091/178466402-1c6ed21f-2bc0-4a43-96a1-93526002f23c.png)
![image](https://user-images.githubusercontent.com/28955091/178466463-aa4dc5a2-8171-440f-bfb5-c8832fa8f351.png)
![image](https://user-images.githubusercontent.com/28955091/178466510-4e2e0feb-03b6-4798-afbd-a3994d52753b.png)
![image](https://user-images.githubusercontent.com/28955091/178466537-088c8f6e-c838-4c07-be13-02e305d403cb.png)

Where, **ridge_1b=1e-6, curvature_2b=1e-4, batch_size=5, weight=0.5**

Besides, I also tested smaller **curvature_2b values**, eg. *1e-2*, even *1e-1*, unfortunately, as shown below (Training:80/200, curvature_2b=1e-1, batch_size=5), it's also the same case. I would appreciate it if you have any clue about that!!
![image](https://user-images.githubusercontent.com/28955091/178467965-80264ae9-dcaf-4444-aa19-da2b2a7d150a.png)

However, the prediction looks perfect, I kinda confused by this.
![image](https://user-images.githubusercontent.com/28955091/178467295-397c436b-cb62-41c0-8dcd-22a565f62559.png)
![image](https://user-images.githubusercontent.com/28955091/178467348-d48912ce-3756-4110-928a-d4ebf5314c67.png)

**Comment by MengnanCui** [comment]
For your information, here is the **pair distribution** of SiC dataset:
![image](https://user-images.githubusercontent.com/28955091/178493674-37adfa7f-2050-4ca8-950d-afa205905a49.png)

**Comment by MengnanCui** [comment]
Hi, another piece of information.
I am trying to repeat the training for NeXe, so I found the example ipynb file, and the dataset "dump.lammpstrj" on my own laptop, and It works very well!!

Afterward, I used **FHI-AIMS** to do a single-point calculation for this NeXe dataset, because, honestly, I am not familiar with **Lammps**. Then I got a new file named "dft.xyz", which contains the same structures, but new energy and forces data, so I re-run the example "ipynb" file again, unfortunately, It came out a weird  Splines picture, as shown below:
![image](https://user-images.githubusercontent.com/28955091/178766230-a85ed724-eb9a-4d05-9f73-f411f2736ba6.png)

It's obvious that there must be something wrong inside, and maybe some details that I neglected,  so I uploaded the new dataset "dft.xyz" and new "pair_potential_demo.ipynb" file [here](https://thcloud.rz-berlin.mpg.de/s/6BFmQGqrBH3p5o9) (where I only edit two places, in order to read from a xyz file), and hope you can have a look recently, since I am struggle on this problem (perfect prediction, but seems unreasonable splines fitting). Probably it's my bad, according to my past experience, I usually did stupid things, well.., but I would appreciate it if you have any clues about it.

**Comment by sxie22** [comment]
Hi @MengnanCui, thanks for sharing the figures and your troubleshooting process. The problem in your Si-C curve is one that we have seen too. It occasionally happens when the lower bound for a particular chemical interaction is lower than the smallest observed distance of that type in the dataset. The curvature regularization is also failing here and it looks like it is simply extrapolating with negative curvature.

The upcoming 0.2.0 release includes a postprocessing feature to adjust the spline coefficients at small coefficients to enforce repulsion. [Feel free to preview that function on our development branch](https://github.com/uf3/uf3/blob/e4b852641ec5161c4aa3ac650d03a8b009dab6f6/uf3/regression/least_squares.py#L608).

The [WIP demo on the development branch](https://github.com/uf3/uf3/blob/e4b852641ec5161c4aa3ac650d03a8b009dab6f6/examples/tungsten_extxyz/pair_potential_WIP.ipynb) includes some of the other upcoming functions which might be helpful.

**Comment by sxie22** [comment]
@MengnanCui I downloaded your Ne-Xe FHI-AIMS dataset and took a look at your jupyter notebook. I reran it with one modification in the "Initialize Basis" section, where I set "offset_1b" to True. In LAMMPS, the energy of an isolated atom is zero; this is not the case for VASP or, apparently, FHI-AIMS. Once I allowed UF3 to fit the one-body terms, the potentials look good.

It looks like the FHI-AIMS reference energies were about -104685 per Ne and -104678 per Xe, which explains why the two-body terms were trying to compensate and became so extreme. The two-body potentials I got had potential wells of about -0.0005.
